st2actions/st2actions/runners/pythonrunner.py

```diff
@@ -158,32 +157,50 @@ def run(self, action_parameters):
         if ACTION_OUTPUT_RESULT_DELIMITER in stdout:
             split = stdout.split(ACTION_OUTPUT_RESULT_DELIMITER)
             assert len(split) == 3
-            result = split[1].strip()
+            action_result = split[1].strip()
             stdout = split[0] + split[2]
         else:
-            result = None
-
+            action_result = None
         try:
-            result = json.loads(result)
+            action_result = json.loads(action_result)
         except:
             pass
+        action_status = None
+
+        if action_result and action_result != "None":
+            result = action_result['result']
+            try:
+                action_status = action_result['status']
+            except:
+                pass
+        else:
+            result = "None"
 
         output = {
             'stdout': stdout,
             'stderr': stderr,
             'exit_code': exit_code,
             'result': result
         }
-
         if error:
             output['error'] = error
-
-        if exit_code == 0:
-            status = LIVEACTION_STATUS_SUCCEEDED
-        elif timed_out:
-            status = LIVEACTION_STATUS_TIMED_OUT
+        status = None
+        if action_status:
```

Comment:
Same here.

I think it could also be simplified a bit (shouldn't need that many if / else branches).